enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,477 @@
+that before the packet is sent to br-ex, it will be DNATed to the service's
+clusterIP by host iptables rule.
+
+```none
+-A OVN-KUBE-NODEPORT -p tcp -m addrtype --dst-type LOCAL -m tcp --dport 30847 -j DNAT --to-destination 10.43.117.48:80
+```
+
+### External to NodePort Services
+
+The general flow is:
+
+1. The packet enters host via eth0.
+2. In the host, the packet is DNATed to the service's clusterIP by the following iptables rule.
+
+```none
+-A OVN-KUBE-NODEPORT -p tcp -m addrtype --dst-type LOCAL -m tcp --dport 30847 -j DNAT --to-destination 10.43.117.48:80
+```
+
+3. The packet is forwarded to br-ex bridge.
+
+The rest of the steps is the same as the normal local gateway mode.
+
+### Workflow Description
+
+#### Deploying
+
+There will be no change to the MicroShift deployment procedure.
+
+#### Upgrading
+
+The microshift-ovs-init.service will be responsible for the OVS bridge change.
+As we will not bind interface to br-ex bridge, the `gatewayInterface` flag
+in `/etc/microshift/ovn.yaml` will be deprecated. Also, as the configuration of
+OVS will not affect host physical interfaces, there is no need to allow user to
+config the OVS bridges. The flag `disableOVSInit` can also be deprecated.
+
+#### Configuring
+
+There is no need to introduce any new configuration flag for this change. Unless
+we want to allow user to specify the IP address used by the br-ex interface.
+
+#### Deploying Applications
+
+Applications are deployed as usual.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+On the OVN-Kubernetes side, we need to change the local gateway mode:
+
+1. Allow using an external gateway bridge without uplink port
+2. If there is no gateway available via br-ex, configure default route via
+   169.254.169.4.
+
+   ```none
+   0.0.0.0/0             169.254.169.4 dst-ip rtoe-GR_ovn-control-plane
+   ```
+
+On the MicroShift side, we will need to make the following change:
+
+1. We need to update the microshift-ovs-init.service to 
+   * Create br-ex bridge, and assign a static IP to it.
+   * Since we don't need NetworkManager to manage the DHCP client. We can remove
+   those logic from the configure-ovs.sh.
+   * For upgrading, we also need to remove the NetworkManager connections
+   created by configure-ovs.sh in previous version.
+
+### Risks and Mitigations
+
+The OVN-Kubernetes configuration described in this document is not used by any
+other known platforms. Therefore there is no CI with this setup in
+OVN-Kubernetes upstream. In order to minimize the risk of regression, we need to
+improve the OVN-Kubernetes CI to cover this setup.
+
+### Drawbacks
+
+The egress features that rely on sending packets to wire without passing kernel
+will not work. As there is no uplink port to forward those packets to. So
+egressGW and egressIP will not work. A possible solution is to forward those
```

Comment:
ah that was my next question... how will `egressIP` work? :) I guess egressSVC will work well since its using the lgw mode style already

Reply:
We don't need those features in Microshift. Yes, I think egressSVC will work. EgressFW can also work if we fixed the bug.